main.rb

```diff
@@ -0,0 +1,120 @@
+module Enumerable
+  # each
+  def my_each
+    each do |i|
+      yield i
+    end
+  end
+
+  # each_with_index
+  def my_each_with_index
+    result = map do |element, index|
+      yield "#{index}: #{element} "
+    end
+    result
+  end
+
+  # select
+  def my_select
+    result = []
+    each do |i|
+      result << i if yield(i)
+    end
+    result
+  end
+
+  # all
+  def my_all?
+    my_each do |_elememt|
+      return false unless yield(element)
+    end
+    true
+  end
+
+  # any
+  def my_any?
+    my_each do |_elememt|
+      return true if yield(element)
+    end
+    false
+  end
+
+  # negated
+  def my_none?
+    my_each do |element|
+      yield(element)
+    end
+  end
+
+  # count
+  def my_count(element = false)
+    return length if elememt == false && !block_given?
+
+    arr = []
+    if element
+      to_a.my_each do |e|
+        arr << e if e == element
+      end
+    else
+      to_a.my_each do |e|
+        arr << e if yield(e)
+      end
+    end
+    arr.length
+  end
+
+  # map
+  def my_map(&procs)
+    arr = []
+    my_each do |element|
+      arr << procs.call(element)
+    end
+    arr
+  end
+
+  # inject iterator
+  def my_inject
+    accmulator = self[0]
+    my_each do |i|
+      accmulator += i
+    end
+    accmulator
+  end
+end
```

Comment:
- [ ] `my_inject` combines all elements of enum by applying a binary operation, specified by a block
`p (1..5).my_inject(4) { |prod, n| prod * n }` should return 480

- [ ] `my_inject` when a symbol is specified combines each element of the collection by applying the symbol as a named method
`p (5..10).my_inject(2, :*)`  should return 302400